weather_dl/download_pipeline/parsers_test.py

```diff
@@ -62,16 +62,150 @@ def test_throws_error(self):
 
 class ParseConfigTest(unittest.TestCase):
 
+    def _assert_not_in_dict_values(self, member, dictionary) -> None:
+        for val in dictionary.values():
+            self.assertNotIn(member, val)
+
     def test_json(self):
-        with io.StringIO('{"key": "value"}') as f:
+        with io.StringIO('{"section": {"key": "value"}}') as f:
             actual = parse_config(f)
-            self.assertDictEqual(actual, {'key': 'value'})
+            self.assertDictEqual(actual, {'section': {'key': 'value'}})
 
     def test_bad_json(self):
-        with io.StringIO('{"key": "value", "brokenKey": }') as f:
+        with io.StringIO('{"section": {"key": "value", "brokenKey": }}') as f:
             actual = parse_config(f)
             self.assertDictEqual(actual, {})
 
+    def test_json_produces_lists(self):
+        with io.StringIO('{"section": {"key": "value", "list": [0, 10, 20, 30, 40]}}') as f:
+            actual = parse_config(f)
+            self._assert_not_in_dict_values('\n', actual['section'])
```

Comment:
Since the arguments are the same for all of these, the helper method can be even more specific: 
```suggestion
            self._assert_no_newlines_in_section(actual)
```

We can make it general when we have a need to.  See the YAGNI principle [1].

[1]: https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it